### ISSUE TITLE
Tray keeping grey and disconnected .

### ISSUE THREAD

**Issue by eromoe** [Martchus/syncthingtray#252]
I get disconnected while syncthing running in bg . Using built-in syncthing, and http://127.0.0.1:8384 is accessable too .

This happend after I solved  https://github.com/Martchus/syncthingtray/issues/250 , in that issue, windows hyper-v took some ports which prevented syncthing bind on 8384. Now the port problem is fixed, but the tray still grey and show disconnected .

**Comment by Martchus** [comment]
Are you sure that reconnects are configured? (Although it is probably also not normal to get disconnected in the first place.)

Sounds like this is a duplicate of https://github.com/Martchus/syncthingtray/issues/217 which I couldn't reproduce so far.

**Comment by eromoe** [comment]
Yes, click this two place many times
![image](https://github.com/Martchus/syncthingtray/assets/3938751/0e4805cb-2974-46f7-a40c-7201706dc81b)
It's not sometimes, but no reaction at all.

**Comment by Martchus** [comment]
True, when clicking the button doesn't help then it is a different problem.

> I don't know where to see the logs, I mean logs of tray .

If there are errors you will get a notification about them - unless you disabled those. You can access errors also by right-clicking on the tray icon and selecting errors.

You can also show logs by running the tray from a terminal. Under Windows you need to use a redirection or the `-cli.exe`-binary to see any output. It would probably be useful to set the env variable `LIB_SYNCTHING_CONNECTOR_LOG_ALL=1` before starting the tray (via the terminal). Also make sure to terminate the other tray process first (as there will normally only be one instance running).

**Comment by eromoe** [comment]
logs is blank : 
![image](https://github.com/Martchus/syncthingtray/assets/3938751/5ecb5fae-e77c-4281-b3a4-7fee6e728a06)

Try -cli.exe , nothing output too
![image](https://github.com/Martchus/syncthingtray/assets/3938751/6574af0f-ee7c-4b32-9746-bb1f3632b60d)

**Comment by Martchus** [comment]
Ok, so really not obvious what's (not) going on here. That it not shows anything means it is either really broken in your environment or that outputting in the terminal doesn't work. To check for the latter you can try to invoke it with `--help`. If that prints nothing then outputting in the terminal doesn't work. I tested that under Windows 10 with `cmd.exe` and the Windows Terminal but if you're using a different terminal it may not work. Then you may try the version without `-cli` and/or a redirection to a file.

You could try to set a low timeout in the connection settings (you'll have to check the option to show the advanced config for that). If it then runs into a timeout we know that it at least tried to connect.

If you're using HTTPs it would make sense to try with HTTP instead. (You'll have to change the setting in Syncthing itself and then the URL in Syncthing Tray.)

If you're using a Windows 10 version older than 1809 you'll have to upgrade or use the Qt 5 based version. It may generally be worth trying the Qt 5 based version (although the Qt 6 based version is supposed to work and normally the better option).

Otherwise I also don't really know what to suggest and cannot really help as it works on my machine (with Windows 10).

**Comment by eromoe** [comment]
Stop syncthing instance then click  `appl connnection setting and try to reconnect with the currently selected config`  would show `reconnecting` . Once start `syncthing instance` , click that button show disconnected .

I tried the qt5 version, is similar (however it mess up some items with Chinese characters in the configuration file , I use a whole english path to test ).

**Comment by eromoe** [comment]
Just find API key in syncthing WEBUI  doesn't match the value in syncthing tray.After copy the value in webui to syncthing tray,  log showing, though the status is keeping in reconnecting...

**Comment by eromoe** [comment]
Oh,  timeouts was set to 1ms but not show if doesn't check `advance settings`  as your recomend testings,  remove the timeout get to work.
So the problem is `API key` doesn't match,  but no error log is unexpecting.

**Comment by Martchus** [comment]
> Oh, timeouts was set to 1ms but not show if doesn't check `advance settings` as your recomend testings, remove the timeout get to work.

That makes sense - although you should have gotten an error notification that the request was running into a timeout.

> … but no error log is unexpecting.

That's true - also for an authentication error you should have gotten an error notification.

Did you perhaps disable notifications for errors (the second checkbox under notification settings)? But even if disabled, the error should have been accessible via the context menu you can open by right-clicking on the tray icon.

---

> @Martchus Is there any easy and safe way to save settings of sync folder only, and try with a fresh config?

That's actually a question about Syncthing itself - and it is a good one. I suppose you could mess around with Syncthing's XML-based config manually (while Syncthing is stopped). I'd have to checkout the XML structure myself but probably not much speaks against copying the folder-elements from an old config file to a new one. You may have to adjust some details like paths, though.
